Real form env setup

dave-mills opened the pull request:
PR to add the ability to "reset" the local environment to have 3 teams that all have deployed versions of the real HOLPA forms, linked to a specific set of ODK Central projects that lets us share on submission making and testing.


To test - follow the instructions in the README.

dan-tang-ssd commented:
I am reviewing this PR.

Just finished testing the first two parts. I will continue the third part tomorrow.

Here is my testing result for progress update.

---

1. Default; no specific ODK database link

It works very well.
There is no xlsform template to check.

---

2. Setup with Mini Test Forms

I tested below items:
 - [x] download xlsform template ODK form excel file
 - [x] download media files
 - [x] check xlsform template content
   -  it seems datasets are mismatched with main survey and repeat groups
   - correction is required in seeder files
 - [x] retrieve submissions from ODK central
   - Non Program Test Team, Test Mini HOLPA Household Survey, 2 out of 3 submissions have been retrieved
     - It should not be an issue related to this PR. I just mark it down as a reference.

---

Screen shots:

![image](https://github.com/user-attachments/assets/b9a5e559-b370-4e2d-8437-09133ead4fa1)

![image](https://github.com/user-attachments/assets/c12a971f-6ef3-4457-812a-4022a1993d5e)

dave-mills commented:
@dan-tang-ssd thanks for fixing the seeder namespacing. I wouldn't worry about whether the datasets are matched up - that's not the focus of this PR.  - the test forms will likely only be used as tests for generic"user creates a brand new dataset and links it to a form" - which will mean we're creating new datasets, and the datasets for the real forms will probably be changed anyway as we explore the data structures more and refactor how XlsformTemplateSection, XlsformModule, Dataset and DatasetTemplate models all work.

dan-tang-ssd commented:
3. Setup with Real HOLPA Forms

Make correction on namespace for 4 seeder files in seeders/TestRealForms folder.
All commands executed without error.

Cannot perform testing on retrieving submissions as there is no submission for real forms on ODK central yet.

I can download the ODK form for real household form successfully, but I got error 404 when downloading the real fieldwork form.
I did a checking on file system, both files existed in correct folder. Not sure why Fieldwork form cannot be downloaded...

Household form download link:
http://holpa-platform.test/storage/1/HOLPA-Household.xlsx

Fieldwork form download link:
http://holpa-platform.test/storage/5/HOLPA-Fieldwork.xlsx

---

Screen shots:

![image](https://github.com/user-attachments/assets/ddf35982-e16e-4414-b207-5e994366c064)

![image](https://github.com/user-attachments/assets/444ec399-84d9-4670-92e2-961318704949)

![image](https://github.com/user-attachments/assets/ccf3d2c7-d953-45c7-bef4-0327f7f734aa)

![image](https://github.com/user-attachments/assets/a4c4a1f7-971c-4803-9554-b2bce6d222dd)

![image](https://github.com/user-attachments/assets/4d55b97d-918d-4c1e-ac13-1ec1ec2b0cf4)

dan-tang-ssd commented:
I tried to download the media files (csv files, png files) from both real forms. All media files cannot be downloaded.

Checked that the files are existed in file system...

Example URL:
https://holpa-platform.test/storage/2/area_units.csv



---

Screen shots:

Real Household Form:
![image](https://github.com/user-attachments/assets/c588e9ea-0e0b-442a-8b35-2f281b41301d)

![image](https://github.com/user-attachments/assets/da8bf952-63b0-4793-8efe-8bca1f3b1740)

Real Fieldwork Form:
![image](https://github.com/user-attachments/assets/e6516c87-6b2e-4209-a688-f33b2d9722d1)

![image](https://github.com/user-attachments/assets/5004fac9-465d-4a43-90b6-b686e9a9421f)

Example media file:
![image](https://github.com/user-attachments/assets/0d10cc69-0e5c-43a8-9f18-4a04e030e658)

dave-mills commented:
> I can download the ODK form for real household form successfully, but I got error 404 when downloading the real fieldwork form.

2 things were happening here: 

1. Some of the entries in the media table were set as using the 'publicI' disk, and I'd forgotten to change them to 'local' storage (which is where we're copying the files into: `storage/app`, instead of `storage/app/public`.  
2. After doing that, I realised the local storage disk didn't have a url setup, so I added that and a basic DownloadController, so that users can access the files that are hidden behind auth. 

Fixed in my latest commit

dan-tang-ssd commented:
@dave-mills , Thanks a lot for the fix.

All xlsform files and media files can be downloaded successfully now.

I think it is good to merge this PR now.

dave-mills commented:
@ciara-mc - PR merged into dev; over to you for the auto-submission generation!
